pkg/devfile/adapters/kubernetes/component/utils.go

```diff
@@ -0,0 +1,38 @@
+package component
+
+import (
+	devfileCommon "github.com/openshift/odo/pkg/devfile/versions/common"
+	"github.com/openshift/odo/pkg/kclient"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func convertEnvs(vars []devfileCommon.DockerimageEnv) []corev1.EnvVar {
+	kVars := []corev1.EnvVar{}
+	for _, env := range vars {
+		kVars = append(kVars, corev1.EnvVar{
+			Name:  *env.Name,
+			Value: *env.Value,
+		})
+	}
+	return kVars
+}
+
+func getResourceReqs(comp devfileCommon.DevfileComponent) corev1.ResourceRequirements {
+	reqs := corev1.ResourceRequirements{}
+	limits := make(corev1.ResourceList)
+	if comp.MemoryLimit != nil {
+		memoryLimit, err := resource.ParseQuantity(*comp.MemoryLimit)
```

Comment:
We're not doing any err handling over here, what happens if the devfile has a Memory Limit that `ParseQuantity()` cannot convert? Do we want to ignore and move on to the next component/container?

Reply:
Hmm, can you clarify what you mean by we're not doing any err handling? It may be too late when I'm looking at this again so I could be missing something obvious but to me it looks like if `ParseQuantity()` cannot parse the value then it will return an error and the check on the following line will only set `memoryLimit` if `err == nil` (the non-error case). So if there was an error then `limits` will be empty. That's effectively going to avoid setting a limit. We could revert to some default value but figuring out a good default value may be tricky.

Reply:
Ok, that is what I wanted to know. If there is an error parsing, we just dont set the memory limit for the container